app/plugin_commands.go

```diff
@@ -102,16 +102,25 @@ func (ch *Channels) unregisterPluginCommands(pluginID string) {
 	ch.pluginCommands = remaining
 }
 
-func (a *App) PluginCommandsForTeam(teamID string) []*model.Command {
+// CommandsForTeam returns all the plugin and product commands for the given team.
+func (a *App) CommandsForTeam(teamID string) []*model.Command {
+	var commands []*model.Command
+
 	a.ch.pluginCommandsLock.RLock()
 	defer a.ch.pluginCommandsLock.RUnlock()
```

Comment:
No need to use `defer`, we can simply unlock after the range for `a.ch.pluginCommands`.